task/clair-scan/0.1/clair-scan.yaml

```diff
@@ -39,8 +70,30 @@ spec:
         imagewithouttag=$(echo $IMAGE_URL | sed "s/\(.*\):.*/\1/" | tr -d '\n')
         # strip new-line escape symbol from parameter and save it to variable
         imageanddigest=$(echo $imagewithouttag@$IMAGE_DIGEST)
+        images_processed_template='{"image": {"pullspec": "'"$IMAGE_URL"'", "digests": [%s]}}'
+        digests_processed=""
+
+        for sha_file in /tekton/home/image-manifest-*.sha; do
+          if [ -e "$sha_file" ]; then
+            arch_sha=$(cat "$sha_file")
+            arch=$(basename "$sha_file" | sed 's/image-manifest-//;s/.sha//')
+            arch_specific_digest="$imagewithouttag@$arch_sha"
+
+            echo "Running clair-action on $arch image manifest."
+            # run the scan for each image manifest in the image index
+            clair-action report --image-ref=$arch_specific_digest --db-path=/tmp/matcher.db --format=quay | tee /tekton/home/clair-result-$arch.json || true
 
-        clair-action report --image-ref=$imageanddigest --db-path=/tmp/matcher.db --format=quay | tee /tekton/home/clair-result.json || true
+            digests_processed+="\"$arch_sha\", "
+          fi
+        done
+
+        # run the scan for the Image Index itself
+        clair-action report --image-ref=$imageanddigest --db-path=/tmp/matcher.db --format=quay | tee /tekton/home/clair-result-image_index.json || true
```

Comment:
I'm just afraid this will produce duplicated results, as index image would resolve into the x86_64 architecture, and that image would be tested twice.

We should check "index image" only when it's not an index image

Reply:
makes sense. done